sql/core/src/main/scala/org/apache/spark/sql/jdbc/DriverQuirks.scala

```diff
@@ -39,33 +39,68 @@ import java.sql.Types
  * if `getJDBCType` returns `(null, None)`, the default type handling is used
  * for the given Catalyst type.
  */
-private[sql] abstract class DriverQuirks {
+abstract class DriverQuirks {
+  def canHandle(url : String): Boolean
   def getCatalystType(sqlType: Int, typeName: String, size: Int, md: MetadataBuilder): DataType
   def getJDBCType(dt: DataType): (String, Option[Int])
 }
 
-private[sql] object DriverQuirks {
+object DriverQuirks {
+
+  private var quirks = List[DriverQuirks]()
+
+  def registerQuirks(quirk: DriverQuirks) : Unit = {
```

Comment:
scala doc here as well.  We should probably cover how president works if multiple implementations claim they `canHandle` a given URL.